src/app-logic/url-handling.js

```diff
@@ -48,46 +48,45 @@ function getDataSourceDirs(
 // "null | void" in the query objects are flags which map to true for null, and false
 // for void. False flags do not show up the URL.
 type BaseQuery = {|
```

Comment:
Note: I changed all these optional properties by the use of a `$Shape` below which looked more correct and easier to use. I thought this would fix the issue I had and that's why I did it, but actually it didn't fix it (the libdef change did) so I can revert this if you think it's better.

Reply:
agreed